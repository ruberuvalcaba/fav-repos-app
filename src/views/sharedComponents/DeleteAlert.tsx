import React, { useRef } from 'react'
import {
  AlertDialog,
  AlertDialogBody,
  AlertDialogFooter,
  AlertDialogHeader,
  AlertDialogContent,
  AlertDialogOverlay,
  Button,
} from '@chakra-ui/react'

interface AlertBarProps {
  isOpen: boolean
  onClose: () => void
  onAccept: () => void
}

const DeleteAlert = ({ isOpen, onClose, onAccept }: AlertBarProps): JSX.Element => {
  const cancelRef = useRef()

  return (
    <AlertDialog isOpen={isOpen} leastDestructiveRef={cancelRef} onClose={onClose}>
      <AlertDialogOverlay>
        <AlertDialogContent>
          <AlertDialogHeader fontSize="lg" fontWeight="bold">
            Deleting repo from collection!
          </AlertDialogHeader>

          <AlertDialogBody>Are you sure? You can't undo this action afterwards.</AlertDialogBody>

          <AlertDialogFooter>
            <Button ref={cancelRef} onClick={onClose}>
              Cancel
            </Button>
            <Button colorScheme="red" onClick={onAccept} ml={3}>
              Delete
            </Button>
          </AlertDialogFooter>
        </AlertDialogContent>
      </AlertDialogOverlay>
    </AlertDialog>
  )
}

export default DeleteAlert
