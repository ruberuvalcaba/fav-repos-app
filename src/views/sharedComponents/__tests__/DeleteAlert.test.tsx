import React from 'react'
import renderer from 'react-test-renderer'
import DeleteAlert from '../DeleteAlert'

it('DeleteAlert component renders correctly', () => {
  const component = renderer.create(
    <DeleteAlert isOpen={true} onClose={() => {}} onAccept={() => {}} />
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
