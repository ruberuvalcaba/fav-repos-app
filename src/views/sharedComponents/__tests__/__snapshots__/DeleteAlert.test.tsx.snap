// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAlert component renders correctly 1`] = `<span />`;
